src/integration/java/io/pinecone/integration/dataPlane/QueryErrorTest.java

```diff
@@ -0,0 +1,164 @@
+package io.pinecone.integration.dataPlane;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.grpc.StatusRuntimeException;
+import io.pinecone.clients.AsyncIndex;
+import io.pinecone.clients.Index;
+import io.pinecone.configs.PineconeConnection;
+import io.pinecone.exceptions.PineconeValidationException;
+import io.pinecone.helpers.RandomStringBuilder;
+import io.pinecone.proto.DescribeIndexStatsResponse;
+import io.pinecone.proto.VectorServiceGrpc;
+import io.pinecone.unsigned_indices_model.QueryResponseWithUnsignedIndices;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static io.pinecone.helpers.BuildUpsertRequest.generateSparseIndicesByDimension;
+import static io.pinecone.helpers.BuildUpsertRequest.generateVectorValuesByDimension;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class QueryErrorTest {
+
+    private static Index index;
+    private static AsyncIndex asyncIndex;
+    private static final int dimension = 3;
+
+    @BeforeAll
+    public static void setUp() throws IOException, InterruptedException {
+        PineconeConnection connectionMock = mock(PineconeConnection.class);
+
+        VectorServiceGrpc.VectorServiceBlockingStub stubMock = mock(VectorServiceGrpc.VectorServiceBlockingStub.class);
+        VectorServiceGrpc.VectorServiceFutureStub asyncStubMock = mock(VectorServiceGrpc.VectorServiceFutureStub.class);
+
+        when(connectionMock.getBlockingStub()).thenReturn(stubMock);
+        when(connectionMock.getFutureStub()).thenReturn(asyncStubMock);
+
+        index = new Index(connectionMock);
+        asyncIndex = new AsyncIndex(connectionMock);
+    }
+
+    @Test
+    public void queryWithVectorAndIdSyncTest() {
+        List<Float> values = generateVectorValuesByDimension(3);
```

Comment:
You can probably use the `dimension` value where we're hard-coding this.

Reply:
Actually I should try to isolate each tests as much as possible, so let me remove the class variable.